documentation/content/en/books/handbook/zfs/_index.adoc

```diff
@@ -1335,11 +1335,13 @@ For systems using EFI to boot, execute the following command:
 
 [source,shell]
 ....
-# gpart bootcode -p /boot/boot1.efifat -i 1 ada1
+# mount_msdosfs /dev/ada1p1 /boot/efi
+# cp /boot/loader.efi /boot/efi/efi/boot/bootx64.efi
```

Comment:
This assumes amd64

Reply:
Fair point. Most of the other FreeBSD docs I've seen *only* copy loader.efi to bootx64.efi, as that's correct likely ~99% of the time. The [wiki page](https://wiki.freebsd.org/UEFI) and [man page](https://man.freebsd.org/cgi/man.cgi?query=uefi) note there is no 32-bit UEFI support, so 'x64' and 'aa64' would cover probably 99.9% of cases.

```sh
# cp /boot/loader.efi /boot/efi/efi/boot/bootx64.efi
or
# cp /boot/loader.efi /boot/efi/efi/boot/bootaa64.efi
```

To cover all cases, perhaps borrow from the convention used in the loader.efi man page?

```sh
# cp /boot/loader.efi /boot/efi/efi/boot/bootXXX.efi (see uefi(8) for values to replace ‘XXX’ with)
```

I don't have enough knowledge to explain why we don't just direct users to *only* copy it to the designated FreeBSD spot:

```sh
# cp /boot/loader.efi /boot/efi/efi/freebsd/loader.efi
```